contracts/MarketOracle.sol

```diff
@@ -52,14 +57,16 @@ contract MarketOracle is Ownable {
 
             volumeWeightedSum = volumeWeightedSum.add(partialRate.mul(partialVolume));
             volumeSum = volumeSum.add(partialVolume);
+            isDataFresh = isDataFresh || isSourceFresh;
         }
 
-        // No explicit fixed point normalization is done as dividing by volumeSum normalizes
-        // to exchangeRate's format.
-        uint256 exchangeRate = volumeSum > 0
-            ? volumeWeightedSum.div(volumeSum)
-            : 0;
-        return (exchangeRate, volumeSum);
+        if (isDataFresh && (volumeSum > 0)) {
```

Comment:
```suggestion
        if (volumeSum > 0) {
```

Reply:
isDataFresh variable isn't needed. if volumeSum > 0 then isDataFresh only possible value is true. if volumeSum == 0  isDataFresh only possible value is false.

Reply:
👍